packages/xml-views-validation/src/validators/attributes/unknown-attribute-key.ts

```diff
@@ -0,0 +1,179 @@
+  if (ui5Class === undefined) {
+    ui5Aggregation = getUI5AggregationByXMLElement(xmlElement, model);
+  }
+
+  // Aggregations don't have additional allowed attributes
+  if (ui5Aggregation !== undefined) {
+    return [
+      {
+        kind: "UnknownAttributeKey",
+        message: `Unknown attribute key: ${attribute.key}`,
+        offsetRange: {
+          start: attribute.syntax.key.startOffset,
+          end: attribute.syntax.key.endOffset,
+        },
+        severity: "error",
+      },
+    ];
+  }
+
+  if (ui5Class === undefined) {
+    return [];
+  }
+
+  // Special case: "class" property can be used on all Controls and selected Elements.
+  // TODO Should this be in the model? Should it be suggested for code completion?
+  if (attribute.key === "class") {
+    return [];
+  }
+
+  // Special case: "binding" property is a synonym for "objectBindings" special setting on ManagedObject
+  // TODO Should this be in the model? Should it be suggested for code completion?
+  if (attribute.key === "binding") {
+    return [];
+  }
+
+  // Special case: "stashed" property can appear under sap.uxap.ObjectPageLazyLoader
+  // Technically it can be under any Element but it should only be used for this one.
+  // TODO Should this be in the model? Should it be suggested for code completion?
+  if (
+    attribute.key === "stashed" &&
+    ui5NodeToFQN(ui5Class) === "sap.uxap.ObjectPageLazyLoader"
+  ) {
+    return [];
+  }
+
+  const allProps: (
+    | UI5Prop
+    | UI5Event
+    | UI5Association
+    | UI5Aggregation
+  )[] = flattenProperties(ui5Class);
+  const allEvents = flattenEvents(ui5Class);
+  const allAssociations = flattenAssociations(ui5Class);
+  // Aggregations can be used as attributes for binding
+  const allAggregations = flattenAggregations(ui5Class);
+  const allClassUI5Attributes = allProps
+    .concat(allEvents)
+    .concat(allAssociations)
+    .concat(allAggregations);
+  // TODO Should we filter out non-public/protected names?
+  if (
+    find(allClassUI5Attributes, (_) => _.name === attribute.key) !== undefined
+  ) {
+    return [];
+  }
+
+  return [
+    {
+      kind: "UnknownAttributeKey",
+      message: `Unknown attribute key: ${attribute.key}`,
+      offsetRange: {
+        start: attribute.syntax.key.startOffset,
+        end: attribute.syntax.key.endOffset,
+      },
+      severity: "error",
+    },
+  ];
+}
+
+function isAttributeNameAlwaysValid(attribute: XMLAttribute): boolean {
+  // This will never happen - it was checked before calling this function
+  /* istanbul ignore if */
+  if (attribute.key === null) {
+    return true;
+  }
+  // xmlns attributes are always valid
+  // TODO "xmlns:" should not return true (change when using new function in xml-utils)
+  if (isXMLNamespaceKey(attribute.key)) {
+    return true;
+  }
+  const { ns, name } = splitAttributeByNamespace(attribute);
+  // sap.ui.dt:designtime is always valid
+  if (ns === "sap.ui.dt" && name === "designtime") {
```

Comment:
This acts as "annotations" in UI5 xml views?

Reply:
It's used to define if a component can be hidden/changed when writing extensible applications (that allow UI adaptation).
It's specified here under "Restrict adaptation for certain controls":
https://sapui5.hana.ondemand.com/#/topic/de9fd55c69af4b46863f5d26b5d796c4